### DISCUSSION TITLE
Can anybody think of any reason why building with Wails would interfere with `go-ole` (ie windows COM)?

### DISCUSSION THREAD

**Discussion by campbellwmorgan** [wailsapp/wails#1677] [General]
#### Background

I've recently added some code that uses the [go-ole](https://github.com/go-ole/go-ole) library to make some windows COM calls.

When I build my project just from the go build function without the flags `desktop,production` or without `dev` my calls work fine (via a cli command). They also work fine when the GUI is running.

However, when I build with wails using the new `-windowsconsole` flag (yes thank you for adding that!!) the initialization of my ole interface fails with `InvalidFunction`.  (Specifically i call https://github.com/go-ole/go-ole/blob/bac9a21098b9908a7260c558a678008c6e4fead4/com.go#L80)

Interestingly the failure only occurs when executing the executable from the command line. If called via the GUI, it works fine

#### Question

Does Wails internally use the windows COM interface for something? Does it hook into any windows interfaces that may be overwriting a reference to something?

**Comment by campbellwmorgan** [comment]
Just to reply to my own comment. I realise this must be because wails also uses `go-ole` which would make sense (here https://github.com/wailsapp/wails/blob/6a07cbd5052b2d2fc3ef9963796d7636111080e4/v2/internal/go-common-file-dialog/cfd/CommonFileDialog_windows.go#L10) 

And i've resolved my issue by just ignoring the error as has been done in wails. So far everything seems to work fine :)

**Reply by leaanthony** [reply]
Thanks for coming back with that and I'm pleased you got it fixed. This sounds like it would be a great addition to our Troubleshooting guide. Would you consider adding a section in that? All you'd need to do is follow this link and start typing 👍  https://github.com/wailsapp/wails/edit/master/website/docs/guides/troubleshooting.mdx

**Reply by campbellwmorgan** [reply]
Sure! Will do it later in the week
